src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Generic.Tests.cs

```diff
@@ -0,0 +1,307 @@
+
+            Assert.False(queue.TryPeek(out _, out _));
+            Assert.Throws<InvalidOperationException>(() => queue.Peek());
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidPositiveCollectionSizes))]
+        public void PriorityQueue_Generic_Peek_PositiveCount(int count)
+        {
+            IReadOnlyCollection<(TElement, TPriority)> itemsToEnqueue = this.GenericIEnumerableFactory(count).ToArray();
+            (TElement element, TPriority priority) expectedPeek = itemsToEnqueue.First();
+            PriorityQueue<TElement, TPriority> queue = new PriorityQueue<TElement, TPriority>();
+
+            foreach (var (element, priority) in itemsToEnqueue)
+            {
+                if (queue.Comparer.Compare(priority, expectedPeek.priority) < 0)
+                {
+                    expectedPeek = (element, priority);
+                }
+
+                queue.Enqueue(element, priority);
+
+                var actualPeekElement = queue.Peek();
+                var actualTryPeekSuccess = queue.TryPeek(out TElement actualTryPeekElement, out TPriority actualTryPeekPriority);
+
+                Assert.Equal(expectedPeek.element, actualPeekElement);
+                Assert.True(actualTryPeekSuccess);
+                Assert.Equal(expectedPeek.element, actualTryPeekElement);
+                Assert.Equal(expectedPeek.priority, actualTryPeekPriority);
+            }
+        }
+
+        [Theory]
+        [InlineData(0, 5)]
+        [InlineData(1, 1)]
+        [InlineData(3, 100)]
+        public void PriorityQueue_Generic_PeekAndDequeue(int initialCapacity, int count)
+        {
+            PriorityQueue<TElement, TPriority> queue = this.GenericPriorityQueueFactory(initialCapacity, count, out var generatedItems);
+
+            var expectedPeekPriorities = generatedItems
+                .Select(x => x.priority)
+                .OrderBy(x => x, queue.Comparer)
+                .ToArray();
+
+            for (var i = 0; i < count; ++i)
+            {
+                var expectedPeekPriority = expectedPeekPriorities[i];
+
+                var actualTryPeekSuccess = queue.TryPeek(out TElement actualTryPeekElement, out TPriority actualTryPeekPriority);
+                var actualTryDequeueSuccess = queue.TryDequeue(out TElement actualTryDequeueElement, out TPriority actualTryDequeuePriority);
+
+                Assert.True(actualTryPeekSuccess);
+                Assert.True(actualTryDequeueSuccess);
+                Assert.Equal(expectedPeekPriority, actualTryPeekPriority);
+                Assert.Equal(expectedPeekPriority, actualTryDequeuePriority);
+            }
+
+            Assert.Equal(expected: 0, queue.Count);
+            Assert.False(queue.TryPeek(out _, out _));
+            Assert.False(queue.TryDequeue(out _, out _));
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void PriorityQueue_Generic_EnqueueRange_IEnumerable(int count)
```

Comment:
Similarly, there should be a test checking for `null` items to enqueue.

Reply:
Rinse and repeat for the second `EnqueueRange` overload.

Reply:
Covered in `PriorityQueue_Generic_Constructor_IEnumerable_null` and `PriorityQueue_Generic_EnqueueRange_Null`.